### PR TITLE
[RELEASE PR for 2.3.0 release]

### PR DESCRIPTION
**Pull request by apeltzer**
## PR checklist

Full Changelog:

## v2.3.0 - 2024-02-23 - Gray Zinc Dalmatian

- [[#307]](https://github.com/nf-core/smrnaseq/pull/307) - Clean up config file and improve output folder structure
- [[#299]](https://github.com/nf-core/smrnaseq/issues/299) - Bugfix for missing inputs in BAM stats (`genome_quant.r`)
- [[#164]](https://github.com/nf-core/smrnaseq/pull/164) - UMI Handling Feature implemented in the pipeline
- [[#302]](https://github.com/nf-core/smrnaseq/pull/302) - Merged in nf-core template v2.11.1
- [[#294]](https://github.com/nf-core/smrnaseq/pull/294) - Fixed contamination screening issues
- [[#309]](https://github.com/nf-core/smrnaseq/pull/309) - Merged in nf-core template v2.12.0
- [[#310]](https://github.com/nf-core/smrnaseq/pull/310) - Removed unnecessarily separate mirtrace subworkflow, now using module instead
- [[#311]](https://github.com/nf-core/smrnaseq/pull/311) - Fix use of FASTP, set `three_prime_adapter` per default
- [[#314]](https://github.com/nf-core/smrnaseq/pull/314) - Add parameters to control publishing of intermediate results
- [[#317]](https://github.com/nf-core/smrnaseq/pull/317) - Fixed issue with bowtie indices directly supplied
- [[#318]](https://github.com/nf-core/smrnaseq/pull/318) - Merged in nf-core template v2.13.0 and pinned nf-validator

### Parameters

| Old parameter | New parameter                |
| ------------- | ---------------------------- |
|               | `--with_umi`                 |
|               | `--umitools_extract_method`  |
|               | `--umitools_method`          |
|               | `--skip_umi_extract`         |
|               | `--umitools_bc_pattern`      |
|               | `--umi_discard_read`         |
|               | `--save_umi_intermeds`       |
|               | `--save_aligned`             |
|               | `--save_aligned_mirna_quant` |
|               | `--save_merged`              |

### Software dependencies

| Dependency    | Old version | New version |
| ------------- | ----------- | ----------- |
| `multiqc`     | 1.15        | 1.20        |
| `edgeR`       | 3.36.0      | 4.0.2       |
| `limma`       | 3.50.0      | 3.58.1      |
| `bioconvert`  | 0.4.3       | 1.1.1       |
| `mirdeep`     | 2.0.1       | 2.0.1.3     |
| `seqkit`      | 2.3.1       | 2.6.1       |
| `fastqc`      | 0.11.4      | 0.12.1      |
| `samtools`    | 1.17        | 1.19.2      |
| `umitools`    | <none>      | 1.1.5       |
| `umicollapse` | <none>      | 1.0.0       |

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/smrnaseq/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, also make a PR on the nf-core/smrnaseq _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [x] Usage Documentation in `docs/usage.md` is updated.
- [x] Output Documentation in `docs/output.md` is updated.
- [x] `CHANGELOG.md` is updated.
- [x] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: nextflow.config [hunk 1]
```diff
@@ -16,19 +16,29 @@ params {
     // Workflow flags
     protocol                    = 'illumina'
 
+    // References
```
Comment (maxulysse) [inline nextflow.config:19]:
```suggestion
```

Comment (maxulysse) [reply nextflow.config:19]: I think one is enough

**Review by maxulysse — APPROVED** [review]
LGTM

**Comment by apeltzer** [comment]
We're still running quite substantial tests on real data, once these are done / successfull will merge

**Comment by christopher-mohr** [comment]
As far as I know, someone with required permissions would also have to remove the checks `EditorConfig` and `Prettier` for `master`. Also due to the impact of the recent changes, existing reviews should probably be updated.

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,50 @@
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## v2.3.0 - 2024-02-22 - Gray Zinc Dalmatian
```
Comment (christopher-mohr) [inline CHANGELOG.md:6]:
```suggestion
## v2.3.0 - 2024-02-23 - Gray Zinc Dalmatian
```

**Review by christopher-mohr — COMMENTED** [review]
Done with first 100 files, will continue later.

**Review by nvnieuwk — APPROVED** [review]
LGTM :)

**Comment by apeltzer** [comment]
Will also wait for @christopher-mohr 's final words. And running a few combination tests locally on our HPC to be safe. These few hours are not going to hurt anymore ;-)

File: .github/workflows/awsfulltest.yml [hunk 3]
```diff
@@ -14,7 +14,10 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: Launch workflow via tower
-        uses: seqeralabs/action-tower-launch@v2
+        uses: seqeralabs/action-tower-launch@922e5c8d5ac4e918107ec311d2ebbd65e5982b3d # v2
+        # TODO nf-core: You can customise AWS full pipeline tests as required
+        # Add full size test data (but still relatively small datasets for few samples)
+        # on the `test_full.config` test runs with only one set of parameters
```
Comment (maxulysse) [inline .github/workflows/awsfulltest.yml:20]: TODO?

File: CHANGELOG.md [hunk 4]
```diff
@@ -60,23 +104,6 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
-|               | `--mirgenedb_mature`     |
-|               | `--mirgenedb_hairpin`    |
-|               | `--contamination_filter` |
-|               | `--rrna`                 |
-|               | `--trna`                 |
-|               | `--cdna`                 |
-|               | `--ncrna`                |
-|               | `--pirna`                |
-|               | `--other_contamination`  |
-
```
Comment (maxulysse) [inline CHANGELOG.md:79]: This is removed from previous version changelog, are we sure we want to do so?

File: CHANGELOG.md [hunk 5]
```diff
@@ -94,20 +121,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
-| Old parameter        | New parameter    |
-| -------------------- | ---------------- |
-| `--conda`            | `--enable_conda` |
-| `--clusterOptions`   |                  |
-| `--publish_dir_mode` |                  |
-
-> **NB:** Parameter has been **updated** if both old and new parameter information is present.
-> **NB:** Parameter has been **added** if just the new parameter information is present.
-> **NB:** Parameter has been **removed** if parameter information isn't present.
```
Comment (maxulysse) [inline CHANGELOG.md:109]: This is removed from previous version changelog, are we sure we want to do so?

Comment (apeltzer) [reply CHANGELOG.md:109]: Ah probably not. I'll check if I can add this back.

File: .github/workflows/awsfulltest.yml (hunk 3 above)
Comment (apeltzer) [reply .github/workflows/awsfulltest.yml:20]: Its a todo to remove this, yes ;-)

**Review by christopher-mohr — APPROVED** [review]
LGTM

**Pull request merged**